hw2/03-count.js

```diff
@@ -1,3 +1,37 @@
 // Add your code here
 
-input.addEventListener('keydown', handleKeyDown);
+const highlightClass = "bg-warning";
+
+const textContainer = document.getElementById("text-container");
+// Restructure the original text container
+const originalTextSpanList = textContainer.innerText.split(" ").map((word) => {
```

Comment:
![duc way](https://github.com/ptmdmusique/cs564-frontend-homework/assets/38127011/27fa8bae-a9e6-412d-95f0-138433a186b4)

![image](https://github.com/ptmdmusique/cs564-frontend-homework/assets/38127011/33c14bfe-46e0-4763-832c-fcf1abe468f9)



Unfortunately, this program does not capture words such as "way", this is because it is followed by a dash and this code does not take that into account. When "way" is input into the program it should be highlighted too. 

A function that might be useful when trying to achieve this feature is the replaceAll() method. It could be used to replace the original dash with a dash that has a space before and after it. That way when split(" ") is called then words that had dashes in between them would be treated as two separate words instead of one.

Also, this program does not highlight words such as "only". This is because words that are right before a period were not considered.

Reply:
hmmm, I think to keep it consistent with the original text, using `split(<some regex for space, dash, dot, etc>)` would be better but I totally see what you mean

Overall, I think this boils down to the business domain logic that we want, the requirement only mentions that I should highlight text without mentioning special case like `.` and `--` so I stick with the simplest version

I totally agree with you that I miss those special cases, but if we go that far, then maybe we should also consider i18n where words in different languages can have special characters in between, or a simpler case, emoji/unicode characters